aws-cpp-sdk-core/source/utils/Directory.cpp

```diff
@@ -63,6 +64,60 @@ namespace Aws
 			return ss.str();
 		}
 
+        Aws::String GetDirectoryName(const char* fileName)
+        {
+            struct duplicate_path_delims {
+                bool operator()(char a, char b) const {
+                    return a == PATH_DELIM && b == PATH_DELIM;
+                }
+            };
+
+            Aws::String pathName = Aws::String(fileName);
+
+            if (pathName.empty()) return ".";
+
+            // Remove duplicate path delimiters
+            pathName.erase(
+                std::unique(pathName.begin(),
+                            pathName.end(),
+                            duplicate_path_delims()),
+                pathName.end());
+
+            const auto lastPathDelim = std::find(pathName.rbegin(),
+                                                 pathName.rend(),
+                                                 PATH_DELIM);
+
+            if (lastPathDelim == pathName.rend())
+            {
+                // No path delimiters found
+                return ".";
+            }
+            else if (lastPathDelim.base() - pathName.begin() == 1)
```

Comment:
I reverted this change because it did not behave as expected. If I used the `pathName.front() == PATH_DELIM`, it will get the wrong output for the `/d/f`. It will return `/`, but the proper output should be `/d`